app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  # def initialize
+  #   update_posts_counter
+  # end
+
+  def update_posts_counter
+    User.find(author_id).update(posts_counter: Post.where(author_id: author.id).count)
+  end
```

Comment:
- [ ] **Optional suggestion**: 
- You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to a database, then make the `update_post_counter` method private:
- 
//  after_save :update_post_counter

    // private

// def update_post_counter
    author.increment!(:posts_counter)
end

Reply:
Okay, Move on